src/Microsoft.AspNet.OData.Shared/DeltaSetOfT.cs

```diff
@@ -0,0 +1,357 @@
+
+                if (instanceAnnotations != null)
+                {
+                    deltaObject.TrySetPropertyValue(deltaObject.InstanceAnnotationsPropertyInfo.Name, instanceAnnotations);
+                }
+            }
+
+            return deltaObject;
+        }
+
+        private void SetProperties(TStructuralType originalObj, Delta<TStructuralType> edmDeltaEntityObject)
+        {
+            foreach (string property in edmDeltaEntityObject.GetChangedPropertyNames())
+            {
+                edmDeltaEntityObject.TrySetPropertyValue(property, _clrType.GetProperty(property).GetValue(originalObj));
+            }
+
+            foreach (string property in edmDeltaEntityObject.GetUnchangedPropertyNames())
+            {
+                edmDeltaEntityObject.TrySetPropertyValue(property, _clrType.GetProperty(property).GetValue(originalObj));
+            }
+        }
+
+        private DeltaDeletedEntityObject<TStructuralType> CreateDeletedEntityForFailedOperation(Delta<TStructuralType> changedObj)
+        {
+            Type type = typeof(DeltaDeletedEntityObject<>).MakeGenericType(changedObj.ExpectedClrType);
+
+            DeltaDeletedEntityObject<TStructuralType> deletedObject = Activator.CreateInstance(type, true, changedObj.InstanceAnnotationsPropertyInfo) as DeltaDeletedEntityObject<TStructuralType>;
+
+            foreach (string property in changedObj.GetChangedPropertyNames())
+            {
+                SetPropertyValues(changedObj, deletedObject, property);
+            }
+
+            foreach (string property in changedObj.GetUnchangedPropertyNames())
+            {
+                SetPropertyValues(changedObj, deletedObject, property);
+            }
+
+            object annValue;
+            changedObj.TryGetPropertyValue(changedObj.InstanceAnnotationsPropertyInfo.Name, out annValue);
+
+            IODataInstanceAnnotationContainer instanceAnnotations = annValue as IODataInstanceAnnotationContainer;
+
+            if (instanceAnnotations != null)
+            {
+                deletedObject.TrySetPropertyValue(changedObj.InstanceAnnotationsPropertyInfo.Name, instanceAnnotations);
+            }
+
+            deletedObject.TransientInstanceAnnotationContainer = changedObj.TransientInstanceAnnotationContainer;
+
+            TryGetContentId(changedObj, _keys, deletedObject);
+            
+            return deletedObject;
+        }
+
+        private static void TryGetContentId(Delta<TStructuralType> changedObj, IList<string> keys, DeltaDeletedEntityObject<TStructuralType> edmDeletedObject)
+        {
+            bool takeContentId = false;
+            for (int i = 0; i < keys.Count; i++)
+            {
+                object value;
+                edmDeletedObject.TryGetPropertyValue(keys[i], out value);
+
+                if (value == null)
+                {
+                    takeContentId = true;
+                    break;
+                }
```

Comment:
```suggestion
                if (edmDeletedObject.TryGetPropertyValue(keys[i], out value))
                {
                    takeContentId = true;
                    break;
                }
```

Reply:
Here we should takecontentid if the value of the key is null